pull_alerts.py

```diff
@@ -113,27 +122,40 @@ def get_formatted_incidents(recent_incidents):
     return formatted_incidents
 
 
+def _tag_incident(incident, tag_stats):
+    tagged = False
+    for idx, tag in enumerate(TAGS):
+        found_tag = any(tag.tag in note for note in incident.notes)
+        if not found_tag:
+            continue
+        tagged = True
+        tag_stats[idx] += 1
+    return tagged
+
+
 def print_stats(all_incidents, include_stats):
     if not include_stats:
         return
-    actionable = 0
-    non_actionable = 0
+
+    stats_table = PrettyTable()
+    stats_table.field_names = ["Incidents", "Number"]
+    stats_table.align["Incidents"] = "l"
+    stats_table.align["Number"] = "r"
+
+    # tag_stats is indexed the same way as TAGS so tag_stats[0] is #a, etc.
+    tag_stats = [0] * len(TAGS)
+
     not_tagged = 0
     for i in all_incidents:
-        if is_actionable(i):
-            actionable += 1
-        elif is_non_actionable(i):
-            non_actionable += 1
-        else:
-            not_tagged += 1
-    print("""# Statistics
-| Incidents            | Number |
-| -------------------- | ------ |
-| Total                | {:6} |
-| Actionable (#a)      | {:6} |
-| Non Actionable (#na) | {:6} |
-| Not Tagged           | {:6} |
-""".format(len(all_incidents), actionable, non_actionable, not_tagged))
+        tagged = _tag_incident(i, tag_stats)
+        not_tagged += 0 if tagged else 1
```

Comment:
nit - `0 + True == 1` :)

Reply:
how about `int(not tagged)`? :laughing:

Reply:
`int(...)` is not needed but OK :)

Reply:
```python
not_tagged = sum(not _tag_incident(i, tag_stats) for i in all_incidents)
```
:D

but that's arguably not readable since it makes it look like the goal is summation rather than updating the stats

Reply:
Settled on `not_tagged += not tagged`